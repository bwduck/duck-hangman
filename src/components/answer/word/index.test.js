import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AnswerWord from './index';

describe('<AnswerWord/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(
      <AnswerWord pickedLetters={['E', 'S']} word="TEST" />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders all the letters', () => {
    const wrapper = shallow(<AnswerWord pickedLetters={['T']} word="TEST" />);
    const letter = wrapper.find('AnswerLetter');
    expect(letter).toHaveLength(4);
  });
});
