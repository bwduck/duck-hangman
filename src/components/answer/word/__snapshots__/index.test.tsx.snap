// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerWord/> renders and matches the snapshot 1`] = `
<div
  className="c-answer-word"
>
  <AnswerLetter
    key="0"
    letter="T"
    show={false}
  />
  <AnswerLetter
    key="1"
    letter="E"
    show={true}
  />
  <AnswerLetter
    key="2"
    letter="S"
    show={true}
  />
  <AnswerLetter
    key="3"
    letter="T"
    show={false}
  />
</div>
`;
